mapstructure.go

```diff
@@ -993,7 +993,7 @@ func (d *Decoder) decodePtr(name string, data interface{}, val reflect.Value) (b
 	// pointer to be nil as well.
 	isNil := data == nil
 	if !isNil {
-		switch v := reflect.Indirect(reflect.ValueOf(data)); v.Kind() {
+		switch v := reflect.ValueOf(data); v.Kind() {
```

Comment:
I'm not entirely sure if this change is correct. I don't understand why `Indirect` was used here before, given that the nulalbility of `data` is important here, not the value to which `data` points. 

Are there any other call sites that may require updates ?